compiler+runtime/src/cpp/jank/aot.cpp

```diff
@@ -0,0 +1,281 @@
+    llvm::errs() << "Using target triple: " << TargetTripleStr << "\n";
+
+    /* The Driver needs the path to the executable (used for finding related tools/resources)
+     * For this example, we'll use argv[0] or a placeholder. This might need adjustment
+     * depending on how/where your program is run relative to clang resources.
+     * TODO: infer clang++ path. Will be OS dependent and will require to infer correct version's path */
+    auto const ClangExecutablePath{ "/usr/bin/clang++" };
+    clang::driver::Driver TheDriver(ClangExecutablePath,
+                                    TargetTripleStr,
+                                    Diags,
+                                    "jank_aot_compilation",
+                                    VFS);
+    TheDriver.setCheckInputsExist(false);
+
+    std::vector<char const *> Args = { strdup(ClangExecutablePath) };
+
+    for(auto const &module : __rt_ctx->loaded_modules_in_order)
+    {
+      auto const &module_path{ module::module_to_path(module) };
+      util::string_builder sb;
+
+      Args.push_back(strdup(relative_to_cache_dir(module_path).c_str()));
+    }
+
+    Args.push_back(strdup(relative_to_cache_dir("jank_main.cpp").c_str()));
+
+    for(auto const &include_dir : include_dirs)
+    {
+      Args.push_back(strdup(util::format("-I{}", include_dir).c_str()));
+    }
+
+    for(auto const &library_dir : library_dirs)
+    {
+      Args.push_back(strdup(util::format("-L{}", library_dir).c_str()));
+    }
+
+    for(auto const &lib : { "-ljank",
+                            // Default libs that Jank depends on
+                            "-lfolly",
+                            "-lstdc++",
+                            "-lm",
+                            "-lzip",
+                            "-lzippp_static",
+                            "-lcrypto",
+                            "-lclang-cpp",
+                            "-lgc",
+                            "-lgccpp",
+                            "-lnanobench",
+                            "-lftxui-component",
+                            "-lLLVM",
+                            "-lftxui-dom",
+                            "-lftxui-screen",
+                            "-lcpptrace",
+                            "-ldwarf" })
+    {
+      Args.push_back(strdup(lib));
+    }
+
+    for(auto const &define : define_macros)
+    {
+      Args.push_back(strdup(util::format("-D{}", define).c_str()));
+    }
+
+    Args.push_back(strdup("-o"));
+    auto const output_filepath{ relative_to_cache_dir(output_filename) };
+    Args.push_back(strdup(output_filepath.c_str()));
+
+    /* Required because of `strdup` usage and need to manually free the memory.
+     * Clang expects C strings that we own. */
+    util::scope_exit const cleanup{ [&]() {
+      for(auto const s : Args)
+      {
+        // NOLINTNEXTLINE(cppcoreguidelines-no-malloc): Needs to free these strings becuase we strdupped them
```

Comment:
The justification here is duplicated from three lines above. We don't need both. Please use `/* */` comments.